samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -349,7 +347,8 @@ static int settings_set(int argc, char **argv, void *val_ctx)
 
 static int init_settings(void)
 {
-	if (IS_ENABLED(CONFIG_SETTINGS)) {
+	if (IS_ENABLED(CONFIG_SETTINGS) &&
```

Comment:
If you only register settings handler when direct advertising is supported, you could do the same for `init_settings()`. Settings are not needed in this module otherwise.

Reply:
discussed offline, my mistake

Reply:
discussed offline